### ISSUE TITLE
[Warning] Array to string conversion

### ISSUE THREAD

**Issue by axllent** [symbiote/silverstripe-gridfieldextensions#344]
PHP produces a warning in relation to PR #325 when `default_sort` is an array, eg:
```
private static $default_sort = [
    '"SortOrder"' => 'ASC',
];
```
`$sortterm .= ($sortterm ? ', ' : '') . $defaultSort;` incorrectly assumes `$default_sort` is always a string ~ it can be either a string or an array.

CC @davejtoews

**Comment by axllent** [comment]
It pains me to come back to this with my head low, however this is not actually a bug at all, more like an undocumented (and only partially working) Silverstripe feature. Although using arrays for `$default_sort` do actually work within Silverstripe, the API documentation and `DataObject:class` source code clearly specify the value should be a string, and nowhere can I (now) find reference to it ever being an array.

The best I can say is, `$default_sort` is not supposed to be an array, even if it does work. I shall now close this issue, and apologies for my mistake.

**Comment by NightJar** [comment]
I think this is valid & should be reopened. Although the DocBlock on DataObject might state that the parameter should be a string, it is a case of classic poorly updated documentation and obtuse APIs.

For instance, the DocBlock on the private static variable states it should be a string, then immediately instantiates it to `null`.

DataQuery::sort is the main place this is used, but the DocBlock there states it expects fully escaped raw SQL as the parameter (likely from purely [historic reasons](https://github.com/silverstripe/silverstripe-framework/blob/2.4.5/core/model/DataObject.php#L2700-L2715)).
It also expects direction as the second parameter, which isn't set (but could be a part of the so-called SQL statement).

It then delegates though SQLSelect::setOrderBy to SQLSelect::addOrderBy - both of which state explicitly they accept quite an accepting set of parameters and has an allowance for the direction.

```php
     * @example $sql->addOrderBy("Column");
     * @example $sql->addOrderBy("Column DESC");
     * @example $sql->addOrderBy("Column DESC, ColumnTwo ASC");
     * @example $sql->addOrderBy("Column", "DESC");
     * @example $sql->addOrderBy(["Column" => "ASC", "ColumnTwo" => "DESC"]);
```

Then contradicting [years](https://github.com/silverstripe/silverstripe-framework/blob/3.0.0/model/SQLQuery.php#L472-L486) of this type of use, DataObjectSchema goes on to [only accept strings](https://github.com/silverstripe/silverstripe-framework/blob/4.1.0/src/ORM/DataObjectSchema.php#L608-L613)... but only since `4.1.0`!

Classic Silverstripe contradictory support due to poorly or undefined APIs - perhaps due to overusing config, which is a large free-form key-value database (a bit like using excel instead of a database). I argue the solution is to make the API better defined, and more in line with other common and _already supported_ use cases.

While this probably isn't the correct repository for the above argument, but I think it is highly expected _this_ module should make allowance for arrays as the `default_sort` DataObject configuration value. I.e. the complaint is valid, even if the Framework _hints_ otherwise (I claim it is historic and now irrelevant).

**Comment by GuySartorelli** [comment]
fixed in [3.5.1](https://github.com/symbiote/silverstripe-gridfieldextensions/releases/tag/3.5.1) and [3.6.1](https://github.com/symbiote/silverstripe-gridfieldextensions/releases/tag/3.6.1)
